lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -78,8 +109,60 @@ public function __construct($directory, $extension = '')
             ));
         }
 
+        $extension = (string) $extension;
         $this->directory = realpath($directory);
-        $this->extension = (string) $extension;
+        $this->extension = $extension ?: $this->extension;
+        $this->fileMode = 0666 & ~umask();
+    }
+
+    /**
+     * Sets the mode that new directories will be created with.
+     * @param int $mode Mode, normally in octal (ex: 0755)
+     */
+    public function setDirectoryMode($mode)
```

Comment:
@kamermans the default umask on all distributions I know of is ``0022``, which means that the directory will be created as ``0755`` unless you explicitly decide to change it. And even with your own config, the umask is still applied for ``mkdir`` calls.

And actually, for my local setup, I'm rather using ACLs to allow both PHP-FPM and my CLI to write to the same cache folders rather than changing permissions